docs/server/upgrading.rst

```diff
@@ -3,170 +3,259 @@
-  :doc:`customization of style sheets and templates
-  </developers/customization>` to adjust your customizations to the correct
-  locations and approach in the new release.
+   (env) $ pootle check
 
-  .. note:: If you have customized the CSS styling or the JavaScript code you
-     will have to run the following commands to update the static assets:
+.. note:: If you are upgrading from a version of Pootle that uses
+   :file:`localsettings.py` then you may want to merge your old custom settings
+   with your :ref:`settings conf file <settings#customizing>` (default location
+   :file:`~/.pootle/pootle.conf`).
 
-     .. code-block:: bash
 
-       (env) $ cd $pootle_dir/pootle/static/js/
-       (env) $ npm install
-       (env) $ npm update
-       (env) $ pootle webpack
-       (env) $ pootle collectstatic --noinput --clear -i node_modules -i *.jsx
-       (env) $ pootle assets build
+.. _upgrading#start-rq:
 
+Start an RQ Worker
+------------------
 
-     ``$pootle_dir`` is the directory where :command:`pip` installed Pootle. Its
-     location depends on your settings for :command:`pip`, but by default it
-     should be :file:`~/.virtualenvs/env/lib/python2.7/site-packages/`.
+Statistics tracking and various other background processes are managed by `RQ
+<http://python-rq.org/>`_.  The :djadmin:`rqworker` command needs to be run
+continuously in order to process the jobs.
 
+If you have not already done so you should
+:ref:`install and start a Redis server <requirements#packages>`.
 
-* Finally, restart your server.
+You can start the worker in the background with the following command:
 
+.. code-block:: bash
 
-.. _upgrading#post-upgrade:
+   (env) $ pootle rqworker &
 
-Post-upgrade adjustments
-------------------------
+In a production environment you may want to :ref:`run RQ workers as services
+<pootle#running_as_a_service>`.
+
+See here for :ref:`further information about RQ jobs in Pootle <rq>`.
+
+
+.. _upgrading#schema-migration:
+
+Migrate your database schema
+----------------------------
+
+Once you have updated your settings you can perform the database schema and
+data upgrade by running. This needs to be done in a few steps:
+
+.. code-block:: bash
+
+   (env) $ pootle migrate accounts 0002 --fake
+   (env) $ pootle migrate pootle_translationproject 0002 --fake
+   (env) $ pootle migrate
+
+
+.. _upgrading#refresh-stats:
+
+Refreshing checks and stats
+---------------------------
+
+You should now update the translation checks and populate the Redis cache with
+statistical data. You will need to have an :ref:`RQ worker running 
+<installation#running-rqworker>` to complete this.
+
+.. code-block:: bash
+
+   (env) $ pootle calculate_checks
+   (env) $ pootle refresh_stats
+
+This command will dispatch jobs to the RQ worker and may take some time.
+
+If you wish to run ``calculate_checks`` and ``refresh_stats`` in the foreground
+without using the RQ worker you can use the :option:`--no-rq` option.
```

Comment:
In 2.7.1 there is a change that requires to run:

```bash
$ redis-cli -n 1 KEYS "*method-cache:Project:cached_dict:*" | xargs redis-cli -n 1 DEL
```

and I am wondering if we should include a notice with that here.